peepingtom/datablocks/simpleblocks/simpleblock.py

```diff
@@ -13,12 +14,16 @@ class SimpleBlock(DataBlock):
 
     Calling __getitem__ on a SimpleBlock will call __getitem__ on its data property and return a view
     """
-    def __init__(self, data=(), **kwargs):
+
+    def __init__(self, data=(), reader_function=None, **kwargs):
         super().__init__(**kwargs)
+        self.reader_function = reader_function
         self.data = data
 
     @property
     def data(self):
+        if callable(self._data):
+            return self._data.compute()
```

Comment:
Why are you checking `callable` rather than the specific type it should be? Not every callable has the `compute()` method.

Reply:
yep you're right, I'll add the check for `DelayedLeaf` I think it was called